src/BloomBrowserUI/bookEdit/toolbox/readers/libSynphony/synphony_lib.js

```diff
@@ -444,9 +444,24 @@ LibSynphony.prototype.getWordsFromHtmlString = function(textHTML, letters) {
     );
     s = XRegExp.replace(s, regex, " ");
 
-    // split into words using Separator and Control characters
-    // (ZERO WIDTH SPACE is a Control charactor.  See http://issues.bloomlibrary.org/youtrack/issue/BL-3933.)
-    regex = XRegExp("[\\p{Z}\\p{C}]+", "xg");
+    // Split into words using Separator and some Control characters
+    // e.g. ZERO WIDTH SPACE is a Control charactor.  See http://issues.bloomlibrary.org/youtrack/issue/BL-3933.
+    // But so is ZERO WIDTH JOINER. See https://issues.bloomlibrary.org/youtrack/issue/BL-7081.
+    // The following variables are Control(format) characters that should split words.
+    // See list at: https://www.compart.com/en/unicode/category/Cf
+    var zeroWidthSplitters = "\u200b\u200c"; // ZERO WIDTH SPACE / ZERO WIDTH NON-JOINER
+    var ltrrtl = "\u200e\u200f"; // LEFT-TO-RIGHT MARK / RIGHT-TO-LEFT MARK
+    var directional = "\u202A-\u202E"; // more LTR/RTL/directional markers
+    var isolates = "\u2066-\u2069"; // directional "isolate" markers
+    regex = XRegExp(
```

Comment:
1) It would help the readability of this if the \\p{Z} and \\p{Cc} were also assigned to variables. I *infer* that the old code had p{C} and that included more than we want but p{Cc} has less than we want. But for maintenance purposes, it would be easy to clarify that.
2) These should be const, not var.